packages/fxa-payments-server/src/components/PayPalButton/index.tsx

```diff
@@ -79,19 +84,26 @@ export const PaypalButton = ({
 
   const onApprove = useCallback(
     async (data: { orderID: string }) => {
+      /* istanbul ignore next */
       try {
-        setTransactionInProgress(true);
-        const { apiCapturePaypalPayment } = {
+        if (setTransactionInProgress) setTransactionInProgress(true);
+        const { apiCapturePaypalPayment, apiUpdateBillingAgreement } = {
           ...apiClient,
           ...apiClientOverrides,
         };
         // This is the same token as obtained in createOrder
         const token = data.orderID;
-        await apiCapturePaypalPayment({
-          idempotencyKey,
-          priceId,
-          token,
-        });
+        if (newPaypalAgreement && priceId) {
+          await apiCapturePaypalPayment({
+            idempotencyKey,
+            priceId,
+            token,
+          });
+        } else {
+          await apiUpdateBillingAgreement({
+            token,
+          });
```

Comment:
I cannot seem to find the test changes related to this.

Reply:
The PayPalButton Test file doesn't have any tests except `Doesn't render the PayPal button if the PayPal script fails to load`. I'm assuming this was initially skipped because of the difficulty of loading the paypal button third party scripts in a test environment. I think if we want to tackle that it should not be within this pull request.

Reply:
I think Barry means that the `apiUpdateBillingAgreement` function should have tests in `apiClient.test.ts`, just like `apiCapturePaypalPayment`.

Reply:
> 
> 
> The PayPalButton Test file doesn't have any tests except `Doesn't render the PayPal button if the PayPal script fails to load`. I'm assuming this was initially skipped because of the difficulty of loading the paypal button third party scripts in a test environment. 

Oh that's right.  hmmm now that we have some branching logic here, I was hope that get tested.  😢 

> I think Barry means that the apiUpdateBillingAgreement function should have tests in apiClient.test.ts, just like apiCapturePaypalPayment.

That's a good point.  👍

Reply:
Tests added for `apiUpdateBillingAgreement`